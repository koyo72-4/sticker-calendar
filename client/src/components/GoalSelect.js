import React from 'react';

export const GoalSelect = ({ goal, handleGoalChange }) => {
    return (
        <div className="y-margin-15">
            <label htmlFor="goal-select">What is your goal? </label>
            <select
                id="goal-select"
                name="goal"
                value={goal}
                onChange={handleGoalChange}
            >
                <option value="">All Goals</option>
                <option value="exercise">Exercise</option>
                <option value="novel">Work on your novel</option>
                <option value="instrument">Practice a musical instrument</option>
                <option value="sweets">Avoid sweets</option>
            </select>
        </div>
    );
};
