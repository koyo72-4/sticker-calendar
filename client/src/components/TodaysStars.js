import React, { useState } from 'react';
import { MONTHS } from '../util/months';
import DayApi from '../util/dayApi';

const dayApi = new DayApi();

export const TodaysStars = ({ goalsArray, setShowTodayAlert, getStarredDays }) => {
    const [ checkedArray, setCheckedArray ] = useState([]);

    const handleSubmit = event => {
		event.preventDefault();
		const today = new Date();

		const starDayObject = {
			year: today.getFullYear(),
			month: MONTHS[today.getMonth()],
			day: today.getDate(),
			goals: checkedArray
		};

		dayApi.addStars(starDayObject)
			.then(getStarredDays)
			.then(() => setShowTodayAlert(true));

		setCheckedArray([]);
	};

	const handleChange = event => {
        const { name } = event.target;
		if (checkedArray.includes(name)) {
			setCheckedArray(checkedArray.filter(item => item !== name));
		} else {
			setCheckedArray([...checkedArray, name]);
		}
	};

    return (
        <form onSubmit={handleSubmit}>
            <fieldset>
                <legend>What have you worked on today?</legend>
                {goalsArray.map(({ name }) => (
                    <>
                        <input
                            type="checkbox"
                            id={name}
                            name={name}
                            checked={checkedArray.includes(name)}
                            onChange={handleChange}
                        />
                        <label htmlFor={name}>{name}</label>
                        <br />
                    </>
                ))}
                <button type="submit">Add to calendar</button>
            </fieldset>
        </form>
    );
};
