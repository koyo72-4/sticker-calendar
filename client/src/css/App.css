label {
    /* overriding Bootstrap */
    display: initial;
}

.container {
    margin: 50px;
}

#year-input {
    width: 45px;
}

.caret-left {
    margin-right: 10px;
}

.caret-right {
    margin-left: 10px;
}

.y-margin-15 {
    margin: 15px 0;
}

.bottom-margin-17 {
    margin-bottom: 17px;
}

.right-margin-20 {
    margin-right: 20px;
}

.left-padding-3 {
    padding-left: 3px;
}

p {
    margin: 0;
}