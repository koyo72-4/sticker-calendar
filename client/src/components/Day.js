import React, { useContext } from 'react';
import { stickerMap } from '../util/stickers';
import DayApi from '../util/dayApi';
import FetchDataContext from '../util/FetchDataContext';
import '../css/Day.css';

const dayApi = new DayApi();

export const Day = ({ day, month, year, starred, stars, goal }) => {
    const { getStarredDays } = useContext(FetchDataContext);

    const handleClick = () => {
		const starDayObject = {
			year,
			month,
			day,
			goal
		};
		const starMethod = starred ? 'addStarToDay' : 'createStarDay';

		dayApi[starMethod](starDayObject)
			.then(getStarredDays);
    };

    return (
        <td>
            <button
                className="dayButton"
                onClick={handleClick}
            >
                {day}
                {(stars && stars.length > 0) &&
                    stars.map(star => (
                        <span className="star">
                            {stickerMap[star._id.sticker][0]}
                            <span className="screen-reader-only">{star._id.sticker}</span>
                        </span>
                    ))
                }
            </button>
        </td>
    );
};
