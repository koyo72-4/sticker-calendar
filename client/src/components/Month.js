import React from 'react';
import { Day } from './Day';
import '../css/Month.css';

export const Month = React.forwardRef((props, ref) => (
    <table
        style={{tableLayout: 'fixed'}}
        className={`${props.index} opacity0`}
        ref={ref}
    >
        <caption>{`Month${props.index}`}</caption>
        <thead>
            <tr>
                <th>Sunday</th>
                <th>Monday</th>
                <th>Tuesday</th>
                <th>Wednesday</th>
                <th>Thursday</th>
                <th>Friday</th>
                <th>Saturday</th>
            </tr>
        </thead>
        <tbody>
            {props.month.map((week, index, month) => {
                return (
                    <tr key={`tr ${month} ${index} ${week}`}>
                        {week.map((day, index, week) => {
                            const starredDay = props.starredDays.find(starredDay => starredDay.day === day);
                            const starred = !!starredDay;

                            let starsToDisplay;
                            if (starred) {
                                if (props.goal) {
                                    if (starredDay.stars.includes(props.goal)) {
                                        starsToDisplay = [props.goal];
                                    } else {
                                        starsToDisplay = [];
                                    }
                                } else {
                                    starsToDisplay = starredDay.stars;
                                }
                            } else {
                                starsToDisplay = [];
                            }

                            return (
                                <Day
                                    key={`td ${week} ${index} ${day}`}
                                    day={day}
                                    month={props.index}
                                    starred={starred}
                                    stars={starsToDisplay}
                                    handleClick={props.handleClick}
                                />
                            );
                        })}
                    </tr>
                );
            })}
        </tbody>
    </table>
));
