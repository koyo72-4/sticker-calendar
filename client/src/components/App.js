import React from 'react';
import { Month } from './Month';
import { populateYear, monthIndexMap } from '../util/months';
import '../css/App.css';

class App extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			year: 2020,
			populatedYear: populateYear(2020),
			starredDays: []
		};

		this.monthRefs = this.state.populatedYear.reduce((refsObject, value, index) => {
			refsObject[`month${index + 1}`] = React.createRef();
			return refsObject;
		}, {});

		const intersectionCallback = (entries, observer) => {
			entries.forEach(entry => {
				const cssClasses = Array.from(entry.target.classList);
				const opacityClass = cssClasses.find(cssClass => cssClass.includes('opacity'));
				if (entry.intersectionRatio === 1.0) {
					entry.target.classList.replace(opacityClass, 'opacity100');
					observer.unobserve(entry.target);
				} else if (entry.intersectionRatio >= 0.75) {
					entry.target.classList.replace(opacityClass, 'opacity50');
				} else if (entry.intersectionRatio >= 0.5) {
					entry.target.classList.replace(opacityClass, 'opacity25');
				} else if (entry.intersectionRatio >= 0.1) {
					entry.target.classList.replace(opacityClass, 'opacity10');
				} else {
					entry.target.classList.replace(opacityClass, 'opacity0');
				}
			});
		};

		this.observer = new IntersectionObserver(intersectionCallback, {
			threshold: new Array(101).fill(0).map((value, index, array) => index / (array.length - 1))
		});

		this.handleYearChange = this.handleYearChange.bind(this);
		this.handleClick = this.handleClick.bind(this);
	}

	handleYearChange(event) {
		const year = parseInt(event.target.value);
		this.setState({ year, populatedYear: populateYear(year) });
	}

	handleClick() {
		console.log('clicked!');
	}

	componentDidMount() {
		fetch('/api/stars')
			.then(response => response.json())
			.then(result => {
				this.setState({ starredDays: result });
			});

		Object.values(this.monthRefs).forEach(value => {
			this.observer.observe(value.current);
		});
	}

	render() {
		const { starredDays, populatedYear, year } = this.state;

		return (
			<div className="container">
				<h1>Sticker Calendar</h1>
				<br/>
				<label htmlFor="year-select">Year: </label>
				<select id="year-select" onChange={this.handleYearChange}>
					<option value="2020">2020</option>
					<option value="2021">2021</option>
					<option value="2022">2022</option>
					<option value="2023">2023</option>
				</select>
				<br/>
				<p>What is your goal?</p>
				<select>
					<option value="exercise">Exercise</option>
					<option value="novel">Work on your novel</option>
					<option value="instrument">Practice a musical instrument</option>
					<option value="sweets">Avoid sweets</option>
				</select>
				<br/>
				<br/>
				{populatedYear.map((month, index) => {
					const monthIndex = index + 1;
					const starredDaysInMonth = starredDays.filter(starredDay => {
						return (
							(year === starredDay.year) &&
							(monthIndexMap.get(monthIndex) === starredDay.month)
						);
					});

					return (
						<Month
							month={month}
							starredDays={starredDaysInMonth}
							handleClick={this.handleClick}
							index={monthIndex}
							key={monthIndex}
							ref={this.monthRefs[`month${index + 1}`]}
						/>
					);
				})}
			</div>
		);
	}
}

export default App;
