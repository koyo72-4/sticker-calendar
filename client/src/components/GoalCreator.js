import React, { useState, useContext } from 'react';
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';
import { stickerMap } from '../util/stickers';
import { useFormField } from '../util/useFormField';
import GoalApi from '../util/goalApi';
import FetchDataContext from '../util/FetchDataContext';
import '../css/App.css';

const goalApi = new GoalApi();

export const GoalCreator = () => {
    const [showModal, setShowModal] = useState(false);
    const handleClose = () => setShowModal(false);
    const handleShow = () => setShowModal(true);
    const goalInput = useFormField('');
    const sticker = useFormField('star');
    const { getGoals } = useContext(FetchDataContext);

    const saveGoal = (name, sticker) => {
		const goalObject = {
			name,
			sticker
		};
		goalApi.createGoal(goalObject)
			.then(getGoals);
	};
    
    return (
        <>
            <Button
                aria-label="Add a new goal"
                style={{marginLeft: "15px"}}
                onClick={handleShow}
            >
                <span aria-hidden="true">+</span>
            </Button>
            <Modal
                show={showModal}
                onHide={handleClose}
            >
                <Modal.Header closeButton>
                    <Modal.Title>New Goal</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <div>
                        <label htmlFor="goal-input">Goal name: </label>
                        <input
                            type="text"
                            name="name"
                            id="goal-input"
                            {...goalInput}
                        />
                    </div>
                    <div>
                        <label htmlFor="sticker-select">Sticker: </label>
                        <select
                            id="sticker-select"
                            name="sticker"
                            {...sticker}
                            className="select-fontawesome"
                        >
                            {Object.values(stickerMap).map(([faComponent, option]) => option)}
                        </select>
                    </div>
                </Modal.Body>
                <Modal.Footer>
                    <Button onClick={() => saveGoal(goalInput.value, sticker.value)}>
                        Save
                    </Button>
                </Modal.Footer>
            </Modal>
        </>
    );
};
