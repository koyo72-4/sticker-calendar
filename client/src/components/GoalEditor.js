import React, { useState, useEffect, useContext } from 'react';
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';
import { stickerMap } from '../util/stickers';
import GoalApi from '../util/goalApi';
import FetchDataContext from '../util/FetchDataContext';
import '../css/App.css';

const goalApi = new GoalApi();

export const GoalEditor = ({ goalsArray }) => {
    const [ showModal, setShowModal ] = useState(false);
    const [ goalsToShow, setGoalsToShow ] = useState(goalsArray);
    const { getGoals } = useContext(FetchDataContext);

    const handleCloseModal = () => setShowModal(false);
    const handleShowModal = () => setShowModal(true);

    const hideGoal = goalName => {
        setGoalsToShow(goalsToShow.filter(({ name }) => name !== goalName));
    };

    const changeSticker = (goal, { target: { value } }) => {
        const updatedGoals = goalsToShow.map(originalGoal => {
            if (originalGoal.name === goal.name) {
                return { ...originalGoal, sticker: value };
            } else {
                return originalGoal;
            }
        });
        setGoalsToShow(updatedGoals);
    };

    const handleEdit = goalsToKeep => {
		const goalsToChange = goalsToKeep.filter(goal => {
			const originalGoal = goalsArray.find(({ name }) => name === goal.name);
			return originalGoal && originalGoal.sticker !== goal.sticker;
		});

		const goalsToDelete = goalsArray.filter(({ name }) =>
			!goalsToKeep.some(goal => goal.name === name));

		if (goalsToDelete.length && goalsToChange.length) {
			goalApi.deleteGoals(goalsToDelete)
				.then(() => goalApi.updateGoals(goalsToChange))
				.then(getGoals);
		} else if (goalsToDelete.length) {
			goalApi.deleteGoals(goalsToDelete)
				.then(getGoals);
		} else if (goalsToChange.length) {
			goalApi.updateGoals(goalsToChange)
				.then(getGoals);
		}
	};

    useEffect(() => {
        setGoalsToShow(goalsArray);
    }, [goalsArray]);

    return (
        <>
            <Button
                variant="outline-primary"
                size="sm"
                onClick={handleShowModal}
            >
                Edit your goals
            </Button>
            <Modal
                show={showModal}
                onHide={handleCloseModal}
            >
                <Modal.Header closeButton>
                    <Modal.Title>Goal Editor</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    {goalsToShow.map((goal, index) => {
                        const originalGoalSticker = goalsArray.find(originalGoal => originalGoal.name === goal.name).sticker;
                        return (
                            <div
                                className="display-flex justify-content-between"
                                key={index}
                            >
                                <div>
                                    <span>{stickerMap[originalGoalSticker][0]}</span>
                                    <span className="left-padding-3">{goal.name}</span>
                                </div>
                                <div>
                                    <label htmlFor="sticker-select">Sticker: </label>
                                    <select
                                        id="sticker-select"
                                        name="sticker"
                                        value={goal.sticker}
                                        onChange={event => changeSticker(goal, event)}
                                        className="select-fontawesome"
                                    >
                                        {Object.values(stickerMap).map(([faComponent, option]) => option)}
                                    </select>
                                    <Button
                                        variant="outline-danger"
                                        size="sm"
                                        onClick={() => hideGoal(goal.name)}
                                    >
                                        <FontAwesomeIcon icon={faTrashAlt} aria-hidden="true" /> Delete Goal
                                    </Button>
                                </div>
                            </div>
                        );
                    })}
                </Modal.Body>
                <Modal.Footer>
                    <Button onClick={() => handleEdit(goalsToShow)}>
                        Save
                    </Button>
                </Modal.Footer>
            </Modal>
        </>
    );
};
