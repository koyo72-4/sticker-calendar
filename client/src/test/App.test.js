import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { App } from '../components/App';

Enzyme.configure({ adapter: new Adapter() });

global.IntersectionObserver = class IntersectionObserver {
	constructor() {}
	observe() {
		return null;
	}
	unobserve() {
		return null;
	}
};

describe('App', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<App />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	test('renders a calendar', () => {
		const component = shallow(<App />);
		expect(component).toMatchSnapshot();
	});
});
