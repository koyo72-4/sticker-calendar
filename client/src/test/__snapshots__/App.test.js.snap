// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a calendar 1`] = `
<div
  className="container"
>
  <h1>
    Sticker Calendar
  </h1>
  <div
    className="display-flex"
  >
    <div>
      <div
        className="display-flex align-items-center"
      >
        <GoalSelect
          goal=""
          goalsArray={Array []}
          handleGoalChange={[Function]}
        />
        <GoalCreator
          goalInputValue=""
          handleGoalInputChange={[Function]}
          handleStickerChange={[Function]}
          saveGoal={[Function]}
          sticker="star"
        />
      </div>
      <GoalEditor
        goalsArray={Array []}
        handleEditGoals={[Function]}
      />
      <ForwardRef
        addOne={[Function]}
        handleSubmit={[Function]}
        handleYearInputChange={[Function]}
        subtractOne={[Function]}
        updateYear={[Function]}
        year={2020}
        yearInputValue="2020"
      />
      <p>
        No stars have yet been achieved this year. You can do it!
      </p>
    </div>
    <div
      style={
        Object {
          "marginLeft": "50px",
        }
      }
    >
      <TodaysStars
        checkedArray={Array []}
        goalsArray={Array []}
        handleTodayChange={[Function]}
        handleTodaySubmit={[Function]}
      />
    </div>
  </div>
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="0"
    month={
      Array [
        Array [
          "",
          "",
          "",
          1,
          2,
          3,
          4,
        ],
        Array [
          5,
          6,
          7,
          8,
          9,
          10,
          11,
        ],
        Array [
          12,
          13,
          14,
          15,
          16,
          17,
          18,
        ],
        Array [
          19,
          20,
          21,
          22,
          23,
          24,
          25,
        ],
        Array [
          26,
          27,
          28,
          29,
          30,
          31,
          "",
        ],
      ]
    }
    monthName="January"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="1"
    month={
      Array [
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          1,
        ],
        Array [
          2,
          3,
          4,
          5,
          6,
          7,
          8,
        ],
        Array [
          9,
          10,
          11,
          12,
          13,
          14,
          15,
        ],
        Array [
          16,
          17,
          18,
          19,
          20,
          21,
          22,
        ],
        Array [
          23,
          24,
          25,
          26,
          27,
          28,
          29,
        ],
      ]
    }
    monthName="February"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="2"
    month={
      Array [
        Array [
          1,
          2,
          3,
          4,
          5,
          6,
          7,
        ],
        Array [
          8,
          9,
          10,
          11,
          12,
          13,
          14,
        ],
        Array [
          15,
          16,
          17,
          18,
          19,
          20,
          21,
        ],
        Array [
          22,
          23,
          24,
          25,
          26,
          27,
          28,
        ],
        Array [
          29,
          30,
          31,
          "",
          "",
          "",
          "",
        ],
      ]
    }
    monthName="March"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="3"
    month={
      Array [
        Array [
          "",
          "",
          "",
          1,
          2,
          3,
          4,
        ],
        Array [
          5,
          6,
          7,
          8,
          9,
          10,
          11,
        ],
        Array [
          12,
          13,
          14,
          15,
          16,
          17,
          18,
        ],
        Array [
          19,
          20,
          21,
          22,
          23,
          24,
          25,
        ],
        Array [
          26,
          27,
          28,
          29,
          30,
          "",
          "",
        ],
      ]
    }
    monthName="April"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="4"
    month={
      Array [
        Array [
          "",
          "",
          "",
          "",
          "",
          1,
          2,
        ],
        Array [
          3,
          4,
          5,
          6,
          7,
          8,
          9,
        ],
        Array [
          10,
          11,
          12,
          13,
          14,
          15,
          16,
        ],
        Array [
          17,
          18,
          19,
          20,
          21,
          22,
          23,
        ],
        Array [
          24,
          25,
          26,
          27,
          28,
          29,
          30,
        ],
        Array [
          31,
          "",
          "",
          "",
          "",
          "",
          "",
        ],
      ]
    }
    monthName="May"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="5"
    month={
      Array [
        Array [
          "",
          1,
          2,
          3,
          4,
          5,
          6,
        ],
        Array [
          7,
          8,
          9,
          10,
          11,
          12,
          13,
        ],
        Array [
          14,
          15,
          16,
          17,
          18,
          19,
          20,
        ],
        Array [
          21,
          22,
          23,
          24,
          25,
          26,
          27,
        ],
        Array [
          28,
          29,
          30,
          "",
          "",
          "",
          "",
        ],
      ]
    }
    monthName="June"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="6"
    month={
      Array [
        Array [
          "",
          "",
          "",
          1,
          2,
          3,
          4,
        ],
        Array [
          5,
          6,
          7,
          8,
          9,
          10,
          11,
        ],
        Array [
          12,
          13,
          14,
          15,
          16,
          17,
          18,
        ],
        Array [
          19,
          20,
          21,
          22,
          23,
          24,
          25,
        ],
        Array [
          26,
          27,
          28,
          29,
          30,
          31,
          "",
        ],
      ]
    }
    monthName="July"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="7"
    month={
      Array [
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          1,
        ],
        Array [
          2,
          3,
          4,
          5,
          6,
          7,
          8,
        ],
        Array [
          9,
          10,
          11,
          12,
          13,
          14,
          15,
        ],
        Array [
          16,
          17,
          18,
          19,
          20,
          21,
          22,
        ],
        Array [
          23,
          24,
          25,
          26,
          27,
          28,
          29,
        ],
        Array [
          30,
          31,
          "",
          "",
          "",
          "",
          "",
        ],
      ]
    }
    monthName="August"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="8"
    month={
      Array [
        Array [
          "",
          "",
          1,
          2,
          3,
          4,
          5,
        ],
        Array [
          6,
          7,
          8,
          9,
          10,
          11,
          12,
        ],
        Array [
          13,
          14,
          15,
          16,
          17,
          18,
          19,
        ],
        Array [
          20,
          21,
          22,
          23,
          24,
          25,
          26,
        ],
        Array [
          27,
          28,
          29,
          30,
          "",
          "",
          "",
        ],
      ]
    }
    monthName="September"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="9"
    month={
      Array [
        Array [
          "",
          "",
          "",
          "",
          1,
          2,
          3,
        ],
        Array [
          4,
          5,
          6,
          7,
          8,
          9,
          10,
        ],
        Array [
          11,
          12,
          13,
          14,
          15,
          16,
          17,
        ],
        Array [
          18,
          19,
          20,
          21,
          22,
          23,
          24,
        ],
        Array [
          25,
          26,
          27,
          28,
          29,
          30,
          31,
        ],
      ]
    }
    monthName="October"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="10"
    month={
      Array [
        Array [
          1,
          2,
          3,
          4,
          5,
          6,
          7,
        ],
        Array [
          8,
          9,
          10,
          11,
          12,
          13,
          14,
        ],
        Array [
          15,
          16,
          17,
          18,
          19,
          20,
          21,
        ],
        Array [
          22,
          23,
          24,
          25,
          26,
          27,
          28,
        ],
        Array [
          29,
          30,
          "",
          "",
          "",
          "",
          "",
        ],
      ]
    }
    monthName="November"
    starredDays={Array []}
  />
  <ForwardRef
    goal=""
    handleClick={[Function]}
    key="11"
    month={
      Array [
        Array [
          "",
          "",
          1,
          2,
          3,
          4,
          5,
        ],
        Array [
          6,
          7,
          8,
          9,
          10,
          11,
          12,
        ],
        Array [
          13,
          14,
          15,
          16,
          17,
          18,
          19,
        ],
        Array [
          20,
          21,
          22,
          23,
          24,
          25,
          26,
        ],
        Array [
          27,
          28,
          29,
          30,
          31,
          "",
          "",
        ],
      ]
    }
    monthName="December"
    starredDays={Array []}
  />
</div>
`;
